java/org/contikios/cooja/Simulation.java

```diff
@@ -696,55 +748,39 @@ public boolean setConfigXML(Element root, boolean quick) throws Exception {
    *          Mote to remove
    */
   public void removeMote(final Mote mote) {
+    invokeSimulationThread(() -> doRemoveMote(mote));
+  }
 
-    /* Simulation is running, remove mote in simulation loop */
-    Runnable removeMote = new Runnable() {
-      @Override
-      public void run() {
-        motes.remove(mote);
-        currentRadioMedium.unregisterMote(mote, Simulation.this);
+  private void doRemoveMote(Mote mote) {
+    motes.remove(mote);
 
-        /* Dispose mote interface resources */
-        mote.removed();
-        for (MoteInterface i: mote.getInterfaces().getInterfaces()) {
-          i.removed();
-        }
-
-        setChanged();
-        notifyObservers(mote);
+    if (currentRadioMedium != null) {
+      currentRadioMedium.unregisterMote(mote, Simulation.this);
+    }
 
-        // Delete all events associated with deleted mote.
-        eventQueue.removeIf(
-          (TimeEvent ev) ->
-            ev instanceof MoteTimeEvent && ((MoteTimeEvent)ev).getMote() == mote);
-      }
-    };
+    /* Dispose mote interface resources */
+    mote.removed();
 
```

Comment:
I think the added newline removes the grouping of the code that disposes the mote interface resources.

Reply:
I will try to restore the previous new lines.